internal/xpkg/cache.go

```diff
@@ -17,95 +17,103 @@ limitations under the License.
 package xpkg
 
 import (
+	"compress/gzip"
 	"io"
 	"os"
 	"sync"
 
-	"github.com/google/go-containerregistry/pkg/name"
-	v1 "github.com/google/go-containerregistry/pkg/v1"
-	"github.com/google/go-containerregistry/pkg/v1/tarball"
 	"github.com/spf13/afero"
 
 	"github.com/crossplane/crossplane-runtime/pkg/errors"
 )
 
 const (
-	errGetNopCache = "cannot get an image from a NopCache"
+	errGetNopCache = "cannot get content from a NopCache"
 )
 
-// A Cache caches OCI images.
+const cacheContentExt = ".gz"
+
+// A Cache caches package content.
 type Cache interface {
-	Get(tag string, id string) (v1.Image, error)
-	Store(tag string, id string, img v1.Image) error
+	Has(id string) bool
+	Get(id string) (io.ReadCloser, error)
+	Store(id string, content io.ReadCloser) error
```

Comment:
Should we clarify what format `content io.ReadCloser` is expected to contain - i.e. a gzipped xpkg compliant YAML stream? In particular the fact that it's gzipped wasn't obvious to me until I read the implementation.

Reply:
The `content` here is actually not gzipped. Content comes into the cache uncompressed, `Store` wraps the file being written to in a gzip writer, and `Get` wraps the file being read in a gzip reader.